--- sources.conf.default	2010-12-13 12:43:25.000000000 +0100
+++ sources.conf	2010-12-13 12:42:52.000000000 +0100
@@ -17,7 +17,7 @@
 #  NOTE: The port command parses source URLs in order and installs the
 #        first occurrance when a port appears in multiple repositories.
 #        So keep "file://" URLs above other URL types.
-
+file:///Users/Shared/dports [nosync]
 
 #  To get the ports tree from the master MacPorts server in California, USA use:
 #      rsync://rsync.macports.org/release/ports/
