*** sources.conf.default	Fri Oct 25 17:12:36 2013
--- sources.conf	Tue Feb 11 13:13:40 2014
***************
*** 18,23 ****
--- 18,25 ----
  # multiple sources, it installs the first occurrence. For local sources
  # to shadow remote ones, "file://" URLs must come before other URLs.
  
+ file:///Users/Shared/dports [nosync]
+ 
  # A list of rsync mirrors is available at
  # https://trac.macports.org/wiki/Mirrors#Portfiles.
  #
